Core/Src/CAN/inverter_can.c

```diff
@@ -15,16 +18,35 @@ static FDCAN_TxHeaderTypeDef TxHeader;
 
 static uint8_t RxData[8];
 static FDCAN_RxHeaderTypeDef RxHeader;
-int16_t datainverter[4];
-uint32_t idinverter;
+uint32_t idInverter;
+uint32_t last_message_arrival_time = 0;
+bool is_there_inverter_comm_error();
 
+void store_value(can_vars_e var_name, int value)
+{
+    dataInverter[var_name] = value;
+}
 
+int get_value(can_vars_e var_name)
+{
+    return dataInverter[var_name];
+}
+
+can_vars_e get_var_name_from_id_and_pos(int id, int pos)
+{
+    #define ENTRY(a,b,c) \
+    if (id == b && pos == c) return a; else
+    VARIABLES;
+    #undef ENTRY
+    return -1;
+}
 
 //função que inicializa a can do inversor, chamada em initializer.c.
 void initialize_inverter_CAN(FDCAN_HandleTypeDef* can_ref) {
 	can_ptr = can_ref;
 	void CAN_inverter_receive_callback(FDCAN_HandleTypeDef*, uint32_t);
 	initialize_CAN(can_ptr, CAN_inverter_receive_callback, &TxHeader);
+	osTimerStart(inverter_can_timerHandle, 2000 / portTICK_PERIOD_MS);
```

Comment:
Entendi o motivo dessa função aqui, mas esse initialize é chamado antes do kernel do RTOS iniciar, então não funciona. 
Pode colocar o `osTimerStart` antes do for da tarefa que ele vai tá. Se usar semaphoro nem vai precisar.